src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/listingrequest/UpdateListingRequestAction.java

```diff
@@ -136,40 +136,85 @@ public UUID getUserTransactionId() {
 
     @Override
     public UUID[] getAssociatedEntitiesIds() {
-        val ids = hearingParts
+        val ids = hearingPartsToAdd
             .stream()
             .map(HearingPart::getId)
             .collect(Collectors.toList());
 
-        ids.addAll(hearingParts
+        ids.addAll(hearingPartsToRemove
             .stream()
-            .filter(Objects::nonNull)
-            .map(HearingPart::getSessionId)
-            .collect(Collectors.toList()));
+            .map(HearingPart::getId)
+            .collect(Collectors.toList())
+        );
+
+        if (hearing.getStatus().getStatus().equals(Status.Listed)) {
+            ids.addAll(hearingPartsToRemove
+                .stream()
+                .filter(Objects::nonNull)
+                .map(HearingPart::getSessionId)
+                .collect(Collectors.toList()));
+        }
 
         ids.add(updateListingRequest.getId());
 
         return ids.stream().toArray(UUID[]::new);
     }
 
-    private void getHearingPartsWithStatus(Status status) {
-        switch (status) {
-            case Listed:
-                hearingParts = hearing.getHearingParts()
-                    .stream()
-                    .filter(hp -> hp.getStatus().getStatus().equals(status))
-                    .sorted(Comparator.comparing(HearingPart::getStart))
-                    .collect(Collectors.toList());
-                break;
-            case Unlisted:
-                hearingParts = hearing.getHearingParts()
-                    .stream()
-                    .filter(hp -> hp.getStatus().getStatus().equals(status))
-                    .collect(Collectors.toList());
-                break;
-            default:
-                break;
+    private void getUnlistedHearingParts() {
```

Comment:
if U're going with getMetdhod then return a value otherwise try more of a command approach, so for example fillWithUnlistedHearingParts - though it's still not the best name, but that kind of separation helps to read the code for others.

Reply:
Done for all 'get' methods